applications/machine_learning/src/modules/usb_state.c

```diff
@@ -8,15 +8,108 @@
 
 #define MODULE usb_state
 #include <caf/events/module_state_event.h>
+#include <caf/events/power_manager_event.h>
 
 #include <logging/log.h>
 LOG_MODULE_REGISTER(MODULE, CONFIG_ML_APP_USB_STATE_LOG_LEVEL);
 
+/** @brief Peer states. */
+enum usb_state {
+	USB_STATE_DISCONNECTED, /**< Cable is not attached. */
+	USB_STATE_POWERED,      /**< Cable attached but no communication. */
+	USB_STATE_ACTIVE,       /**< Cable attached and data is exchanged. */
+	USB_STATE_SUSPENDED,    /**< Cable attached but port is suspended. */
+	USB_STATE_ERROR         /**< USB error */
+};
+
+
+/* Current state of the USB */
+static enum usb_state current_state = USB_STATE_DISCONNECTED;
```

Comment:
We could limit visibility of the variable to device_status function (since it's only used there)

Reply:
ok